presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1500,7 +1500,13 @@ private void executeIrreversibleMetastoreOperations()
                 }
                 Joiner.on("; ").appendTo(message, failedIrreversibleOperationDescriptions);
 
-                PrestoException prestoException = new PrestoException(HIVE_METASTORE_ERROR, message.toString());
+                PrestoException prestoException;
+                if (suppressedExceptions.size() == 1 && suppressedExceptions.get(0) instanceof TableNotFoundException) {
```

Comment:
@viczhang861 I feel that this logic is based on an assumption that this code may not be in a position to make. E.g. this would break if the caller code changes and it seems unlikely that the caller is aware about this logic. If that's not the case, would you explain why this assumption is valid?

Reply:
Based on comment in caller of this function, this function is only used for delete operation and caller is aware of possible failure
 > "// The next section will deal with "dropping table/partition"." . Commit may still fail in this section.
Does this answer your question?

Reply:
Exploring suppressed exceptions seems very fragile to me as well. How sure we are there's no better way of detecting this exception? Maybe it should be replaced with more specific exception on the level below?

Reply:
The suppressed exceptions inside this function are not ones extracted from Exception.getSuppressed() . I rearranged code to avoid the confusion.